web/src/main/java/org/fao/geonet/Geonetwork.java

```diff
@@ -558,6 +570,39 @@ private boolean checkDBWrite() {
         scheduledExecutorService.scheduleWithFixedDelay(DBHeartBeat, initialDelay, fixedDelay, TimeUnit.SECONDS);
     }
 
+    private void createLinkMonitor(final ResourceManager rm, final GeonetContext gc, String linkMonitorClass,
```

Comment:
I think that this method could be re-written to be a lot more high-level (and therefore, IMO, more readable), e.g.:

```java
private void scheduleLinkMonitor(args...) {
    LinkMonitorInterface linkMonitor = LinkMonitorInterface.newInstance(args...);
    scheduledExecutorService.scheduleWithFixedDelay(linkMonitor, ...);
}
```

This assumes:

* factory method on `LinkMonitorInterface` (i.e. `newInstance()`)
* `LinkMonitorInterface` implements `Runnable` (and can therefore be passed to the executor)
* a shared executor (*probably* only need to the one thread pool between this and db heartbeat, depending on the frequency)

You can choose to do all or none of the above, or somewhere in between. :smile: